src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs

```diff
@@ -219,15 +219,10 @@ public virtual void Contextualize([NotNull] ViewContext viewContext)
         }
 
         /// <inheritdoc />
-        public HtmlString AntiForgeryToken()
+        public HtmlString AntiforgeryToken()
         {
-            var tagBuilder = _htmlGenerator.GenerateAntiForgery(ViewContext);
-            if (tagBuilder == null)
-            {
-                return HtmlString.Empty;
-            }
-
-            return tagBuilder.ToHtmlString(TagRenderMode.SelfClosing);
+            var html = _htmlGenerator.GenerateAntiforgery(ViewContext);
+            return html ?? HtmlString.Empty;
```

Comment:
Why not do the `HtmlString.Empty` thing in `DefaultHtmlGenerator`? No big deal if the tag helper gets `PostContent` a bit more often.